<launch>
    <!-- IMU NODE -->
    <node pkg="barc" type="imu_data_acquisition.py" name="imu_node" >
        <param name="port" value="/dev/ttyACM0" />
    </node>

    <!-- ARDUINO NODE -->
    <node pkg="rosserial_python" type="serial_node.py" name="serial_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node >

    <!-- ODROID TO ARDUINO HUB -->
    <node pkg="barc" type="arduino_interface.py" name="arduino_interface" output="screen">
    </node>
       

    <!-- STATE ESTIMATOR -->
    <node pkg="barc" type="state_estimation_KinBkMdl.py" name="state_estimation" output="screen">
        <param name="L_a" type="double" value="0.125" />
        <param name="L_b" type="double" value="0.125" />

        <!-- v_x estimation sample time-->
        <param name="dt_v_enc" type="double" value="0.2" />

        <!-- ekf properties -->
        <param name="q_std" type="double" value="0.1" />
        <param name="r_std" type="double" value="0.1" />
    </node>    
    
    <!-- Record the experiment data -->
    <!-- <node pkg="rosbag" type="record" name="rosbag_record" args=" -a" /> -->


</launch>
